api/STRUCTURE.md

```diff
@@ -0,0 +1,84 @@
+# All of Us Workbench API Structure & Technologies
+
+The _All of Us_ Workbench API is largely built on [Spring Boot](https://spring.io/projects/spring-boot). It also uses [Swagger](https://swagger.io/) for autogenerating REST APIs.
+
+## Database
+
+The primary application database is a [Google Cloud SQL](https://cloud.google.com/sql/) instance running MySQL. We modify its schema using [Liquibase](https://www.liquibase.org/) migrations, which live in [db/changelog/](https://github.com/all-of-us/workbench/tree/master/api/db/changelog).
+
+When creating a new database migration, add an XML file to the changelog directory that describes the migration to perform. This new file should be named `db.changelog-###-description-of-migration.xml`, where ### is the number of the newest migration + 1. Then, add the file to the end of the [db.changelog-master.xml](https://github.com/all-of-us/workbench/blob/master/api/db/changelog/db.changelog-master.xml) file.
+
+We generally name database objects in lower_snake_case.
+
+Liquibase migrations are applied (indirectly) via a bash script called [db/run-migrations.sh](https://github.com/all-of-us/workbench/blob/master/api/db/run-migrations.sh). Do not run this script directly; instead, call one of the following `project.rb` commands:
```

Comment:
Not necessarily worth noting in here, but TIL that for our GAE environments, the Liquibase migrations are actually executed using the Liquibase Gradle plugin: https://github.com/all-of-us/workbench/blob/331b1228bb7c7b8d392f2acf393c64d82e24a2ef/api/db/build.gradle#L16-L16